webpack/assets/javascripts/react_app/components/SearchBar/SearchAutocomplete.js

```diff
@@ -0,0 +1,166 @@
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { Menu, MenuContent, Popper, SearchInput } from '@patternfly/react-core';
+import classNames from 'classnames';
+import { AutoCompleteMenu } from './AutoCompleteMenu';
+import { translate as __ } from '../../common/I18n';
+import './SearchBar.scss';
+
+export const SearchAutocomplete = ({
+  results,
+  onSearchChange,
+  value,
+  onSearch,
+  disabled,
+  error,
+}) => {
+  const [isAutocompleteOpen, setIsAutocompleteOpen] = useState(false);
+  const searchWrapperRef = useRef(null);
+  const searchInputRef = useRef(null);
+  const autocompleteRef = useRef(null);
+
+  const _onSearch = searchValue => {
+    setIsAutocompleteOpen(false);
+    onSearch(searchValue);
+  };
+  const onClear = () => {
+    onSearchChange('');
+  };
+
+  const onChange = (newValue, e) => {
+    if (
+      searchInputRef &&
+      searchInputRef.current &&
+      searchInputRef.current.contains(document.activeElement)
+    ) {
+      setIsAutocompleteOpen(true);
+      onSearchChange(newValue);
+    }
+  };
+
+  const onSelect = (e, itemId) => {
+    e.stopPropagation();
+    if (itemId[itemId.length - 1] !== ' ') {
+      itemId = `${itemId} `;
+    }
+    if (itemId[0] === ' ') {
+      itemId = itemId.slice(1);
+    }
+    onSearchChange(itemId);
+    searchInputRef.current.focus();
+  };
+
+  const handleMenuKeys = event => {
+    // keyboard shortcut to focus the search, will not focus if the key is typed into an input
+    if (event.key === '/' && event.target.tagName !== 'INPUT') {
```

Comment:
This is a lot of nested `if`s..
if
else if
  if
  else if
  else if

Can you find a way to extract to smaller functions and maybe change to `switch` statements?

Reply:
trimmed the conditions a bit, since every if statement is only 2-3 lines im not sure we should extract it somewhere, and since the conditions are very different I dont a switch is a good fit here